accession-commons-monotonic-generator-jpa/src/test/java/uk/ac/ebi/ampt2d/commons/accession/core/BasicMonotonicAccessioningWithAlternateRangesTest.java

```diff
@@ -0,0 +1,179 @@
+        objects.put("hash4", TestModel.of("service-test-4"));
+        objects.put("hash5", TestModel.of("service-test-5"));
+        objects.put("hash6", TestModel.of("service-test-6"));
+        DecoratedAccessionGenerator<TestModel, Long> generator =
+                DecoratedAccessionGenerator.buildPrefixSuffixAccessionGenerator
+                        (getGenerator(CATEGORY_ID, INSTANCE_ID), "RS", null, Long::parseLong);
+        List<AccessionWrapper<TestModel, String, String>> generated = generator.generateAccessions(objects);
+        assertEquals(6, generated.size());
+        assertEquals("RS1", generated.get(0).getAccession());
+        assertEquals("RS2", generated.get(1).getAccession());
+        assertEquals("RS3", generated.get(2).getAccession());
+        assertEquals("RS4", generated.get(3).getAccession());
+        assertEquals("RS5", generated.get(4).getAccession());
+        assertEquals("RS11", generated.get(5).getAccession());
+    }
+
+    @Test
+    public void testAlternateRangesWithDifferentGenerators() throws AccessionCouldNotBeGeneratedException {
+
+        List<AccessionWrapper<TestModel, String, Long>> evaAccessions = getAccessioningService1(CATEGORY_ID_2, INSTANCE_ID)
+                .getOrCreate(getObjectsForAccessionsInRange(1, 10));
+        assertEquals(9, evaAccessions.size());
+        assertEquals(1, evaAccessions.get(0).getAccession().longValue());
+        assertEquals(9, evaAccessions.get(8).getAccession().longValue());
+        assertEquals(1, contiguousIdBlockService
+                .getUncompletedBlocksByCategoryIdAndApplicationInstanceIdOrderByEndAsc(CATEGORY_ID_2, INSTANCE_ID)
+                .size());
+
+        //get another service for same category
+        evaAccessions = getAccessioningService1(CATEGORY_ID_2, INSTANCE_ID)
+                .getOrCreate(getObjectsForAccessionsInRange(10, 30));
+        assertEquals(20, evaAccessions.size());
+        //Block ended here
+        assertEquals(10, evaAccessions.get(0).getAccession().longValue());
+        //New Block with 200 values interval
+        assertEquals(211, evaAccessions.get(1).getAccession().longValue());
+        assertEquals(220, evaAccessions.get(10).getAccession().longValue());
+        assertEquals(421, evaAccessions.get(11).getAccession().longValue());
+        assertEquals(429, evaAccessions.get(19).getAccession().longValue());
+        assertEquals(1, contiguousIdBlockService.getUncompletedBlocksByCategoryIdAndApplicationInstanceIdOrderByEndAsc
+                (CATEGORY_ID_2, INSTANCE_ID).size());
+
+        //get another service for same category but different Instance
+        evaAccessions = getAccessioningService1(CATEGORY_ID_2, INSTANCE_ID_2)
+                .getOrCreate(getObjectsForAccessionsInRange(30, 39));
+        assertEquals(9, evaAccessions.size());
+        assertNotEquals(430, evaAccessions.get(0).getAccession().longValue());
+        assertEquals(631, evaAccessions.get(0).getAccession().longValue());
```

Comment:
I was going to say that these 2 asserts are redundant, but then I think they make a good point about what must and must **not** happen :wink:

Reply:
yup:)